packages/big-design/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,103 @@
+import { ArrowDropDownIcon, ChevronLeftIcon, ChevronRightIcon } from '@bigcommerce/big-design-icons';
+import React, { useEffect, useState } from 'react';
+
+import { MarginProps } from '../../mixins';
+import { Dropdown } from '../Dropdown';
+import { Flex } from '../Flex';
+
+import { StyledButton } from './styled';
+
+export interface PaginationProps extends MarginProps {
+  currentPerPage: number;
+  currentPage: number;
+  numOfItems: number;
+  perPageOptions: number[];
+  onPageUpdate(page: number): void;
+  onRangeUpdate(range: number): void;
+}
+
+export const Pagination: React.FunctionComponent<PaginationProps> = props => {
+  const { currentPerPage, currentPage, numOfItems, perPageOptions, onPageUpdate, onRangeUpdate } = props;
+
+  const [maxPages, setMaxPages] = useState(Math.ceil(numOfItems / currentPerPage));
+  const [itemRange, setItemRange] = useState([0, 0]);
+
+  useEffect(() => {
+    const first = currentPerPage * (currentPage - 1) + 1;
+    let second = currentPerPage * currentPage;
+    if (second > numOfItems) {
+      second = numOfItems;
+    }
+    const range = [first, second];
+    setItemRange(range);
```

Comment:
You could get a bit more cleaner by removing the `const` declaration
```suggestion
    setItemRange([first, second]);
```